gengokit/httptransport/templates/client.go

```diff
@@ -212,12 +212,25 @@ func contextValuesToHttpHeaders(keys []string) httptransport.RequestFunc {
 	// error and attempt to decode the specific error message from the response
 	// body. Primarily useful in a client.
 	func DecodeHTTP{{$method.Name}}Response(_ context.Context, r *http.Response) (interface{}, error) {
+		buf, err := ioutil.ReadAll(r.Body)
+		if len(buf) == 0 {
+			return nil, errors.New("response http body empty")
+		}
+
+		if err != nil {
```

Comment:
Wow, great catch! Somehow this order happened, I am a little upset that this slipped by me.